### PR TITLE
Adjust Sockets test ports s.t. APM and Async tests do not collide.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@CIPop @davidsh @stephentoub PTAL. In my private testing (e.g. http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/2614/), there have been problems with ports colliding when the APM and async tests run in parallel.

File: src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs [hunk 1]
```diff
@@ -8,7 +11,7 @@ namespace System.Net.Sockets.Tests
 {
     public class AcceptAsync
     {
-        private const int TestPortBase = 8000;
+        private const int TestPortBase = 7000;
```
Comment (stephentoub) [inline src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs:14]: How come this and the subsequent files differ from the previous ones, using a fixed number rather than getting a value from TestPortBases?

**Comment by stephentoub** [comment]
Other than the comment I've made on previous PRs about using dynamically generated port numbers instead (:wink:), LGTM.

File: src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs (hunk 1 above)
Comment (pgavlin) [inline src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs:14]: Two reasons:
1. TestPortBases exists strictly to accommodate the fact that the same sources are built twice into two different assemblies that may run in parallel (and thus collide). These sources do not share this characteristic.
2. Most/all of this manual port management should disappear in the future by transitioning to a plan wherein the OS automatically assigns port numbers. Despite the changes I'm making here, this may need to happen sooner rather than later in order to allow two test runs to happen on the same machine. If I can't figure this out today, the only other option is to disable the Linux tests in the lab for the time being.

File: src/Common/tests/System.Net/Capability.cs [hunk 2]
```diff
@@ -1,145 +0,0 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (CIPop) [inline src/Common/tests/System.Net/Capability.cs:1]: Why was this deleted from sn2?

These files are still needed by other contracts what at this point have no testing. Please leave files in place even if they are not used by sockets. I'd rather not do the work again to port things from TFS to GitHub.

File: src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs (hunk 1 above)
Comment (CIPop) [inline src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs:14]: Some tests will always use allocated ports. We could envision a retry mechanism but never completely remove all tests from allocating their own ports.

File: src/Common/tests/System.Net/Capability.cs (hunk 2 above)
Comment (pgavlin) [inline src/Common/tests/System.Net/Capability.cs:1]: Sorry, that was accidental. I can restore the file.

File: src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs (hunk 1 above)
Comment (pgavlin) [inline src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs:14]: > Some tests will always use allocated ports.

By "allocated" do you mean "chosen ahead of time"? If so, then we'll have to choose those tests very carefully or potentially disable them in CI if it's a requirement that a test must be able to run in parallel with itself (e.g. the same test may be running in multiple processes).

File: src/Common/tests/System.Net/Capability.cs (hunk 2 above)
Comment (CIPop) [inline src/Common/tests/System.Net/Capability.cs:1]: Np: I can add this back. For the future, please leave files in place until we've finished all work.
For the merge with corefx, we can pick only the files that are required by build for the PR. That means that this file shouldn't be picked.

In parallel, we're working on this capability system for System.Net.Http. I think this will end up split in multiple partial classes similar to Capability.Sockets.cs.

File: src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs (hunk 1 above)
Comment (CIPop) [inline src/System.Net.Sockets/tests/FunctionalTests/AcceptAsync.cs:14]: Yes: chosen ahead of time. We could also try a port range and make the tests retry on port reuse failure until the range has been exhausted.

**Comment by CIPop** [comment]
LGTM
